### ISSUE TITLE
How to install into $HOME or `/opt/debops`.

### ISSUE THREAD

**Issue by htgoebel** [debops/debops-tools#59]
For keeping the roles and playbooks up-to-date, I would prefer to install debops into $HOME or into something like `/opt/debops` (which is then owned by the admin-group). In the scripts I've seen some kind of preparation for this, but I could not figure out how it is meant to be used.

Please update the README and/or Makefile for how to achieve this. Thanks.

**Comment by drybjed** [comment]
Here's a quick basic guide, assuming that you want to use `debops-*` scripts. You can either install playbooks and roles system-wide, by cloning https://github.com/debops/debops repository and running `sudo make install`, or just the scripts, by running `sudo make install-scripts`. In the first case, DebOps will be installed automatically in `/usr/local/share/debops` and you will need to run `sudo debops-update` to update roles and playbooks.

If you choose not to install DebOps system-wide, but just the scripts, you can then run `debops-update` (without `sudo`). If you don't specify any directory as target, roles and playbooks will be installed in `~/.local/share/debops/` directory. If you specify a directory as the argument to `debops-update`, it will install/update the roles and playbooks in that directory. Running `debops-update` again will update everything in `~/.local/share/debops/`.

Of course `debops` script knows about all of this and will look for playbooks and roles in `~/.local/share/debops/` and `/usr/local/share/debops/`. Additionally, `debops` works from your current `$PWD` directory, which can be created using `debops-init` script. In your current "project' directory, you can create `playbooks/` and `roles/` and they will take precedence over other roles elsewhere. I use this to develop DebOps roles - I have a separate project directory with my development environment, and when I work on a role, I symlink my own fork to `roles/` directory there. This way, when I need to change something in my production envrionment, it uses "system-wide" roles automatically.

**Comment by JustAnotherGuy** [comment]
The instructions in
http://docs.debops.org/en/latest/installation.html
are currently broken in that they suggest the possibility to install into /opt/debops but the command given is overridden by a non-conditional assignment in Makefile and very suprisingly for the unsuspecting the install takes place into /usr/local/ anyway ~~and~~ ~~of~~ ~~course~~ ~~there~~ ~~is~~ ~~no~~ ~~"make~~ ~~uninstall"~~... [EDIT: use "make clean"]

From drybjed's comment above I gather that installing into /opt/debops won't do if one wants to use the scripts, so someone capable of doing so should update the docs, please.

PS: To fix the Makefile behaviour the below patch could be used, i.e. in case user wants to just install debops to whichever PREFIX but not use the scripts (unless they are also fixed to not use hard-coded paths).

HTH.

``` diff
diff --git a/Makefile b/Makefile
index da05ed8..1302131 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@
 # Part of the DebOps project - http://debops.org/


-PREFIX="/usr/local"
+PREFIX?="/usr/local"
 BIN_DIR="${PREFIX}/bin"
 SHARE_DIR="${PREFIX}/share/debops"

```

**Comment by htgoebel** [comment]
Thanks for reporting and the patch.
I'm not a make-specialist but since I can not remember to have seen `?=` in a Makefile, I did some tests, using this simple Makefile:

```
PREFIX="/usr/local"
all:
    @echo ${PREFIX}
```

And here are the results:

```
$ PREFIX=/asdhfjka make 
/usr/local
$ make  PREFIX=/asdhfjka
/asdhfjka
```

The later should work not only for GNU make, if I read the [POSIX documentation](http://pubs.opengroup.org/onlinepubs/009695399/utilities/make.html) correctly. I just added two pull-requests (#64 reverting your change and https://github.com/debops/docs/pull/29 fixing the docs.)

Many thanks for pointing to this issue!
